frontend/src/metabase/admin/databases/components/DatabaseEditApp/Sidebar/Sidebar.unit.spec.tsx

```diff
@@ -163,6 +163,75 @@ describe("DatabaseEditApp/Sidebar", () => {
     });
   });
 
+  describe("model actions control", () => {
+    it("is shown if database supports actions", () => {
+      setup();
+      expect(screen.getByLabelText(/Model actions/i)).toBeInTheDocument();
+    });
+
+    it("is shown for non-admin users", () => {
+      setup({ isAdmin: false });
+      expect(screen.getByLabelText(/Model actions/i)).toBeInTheDocument();
+    });
+
+    it("shows if actions are enabled", () => {
+      setup({
+        database: createMockDatabase({
+          settings: { "database-enable-actions": true },
+        }),
+      });
+
+      expect(screen.getByLabelText(/Model actions/i)).toBeChecked();
+    });
+
+    it("shows if actions are disabled", () => {
+      setup({
+        database: createMockDatabase({
+          settings: { "database-enable-actions": false },
+        }),
+      });
+
+      expect(screen.getByLabelText(/Model actions/i)).not.toBeChecked();
+    });
+
+    it("isn't shown if database doesn't support actions", () => {
+      const features = _.without(COMMON_DATABASE_FEATURES, "actions");
+      setup({ database: createMockDatabase({ features }) });
+
+      expect(screen.queryByText(/Model actions/i)).not.toBeInTheDocument();
+    });
+
+    it("isn't shown when adding a new database", () => {
+      setup({ database: createMockDatabase({ id: undefined }) });
+      expect(screen.queryByText(/Model actions/i)).not.toBeInTheDocument();
+    });
+
+    it("enables actions", () => {
+      const { database, updateDatabase } = setup();
+
+      userEvent.click(screen.getByLabelText(/Model actions/i));
```

Comment:
Can you add a check before clicking the action that the checkbox isn't checked. So it's easier to follow what's the previous state, and also add the check after clicking the button that not only do we pass correct data to the callback, but the UI is also updated.

Reply:
I personally try to follow [the AAA test structure (Arrange, Act, Assert)](https://medium.com/@pjbgf/title-testing-code-ocd-and-the-aaa-pattern-df453975ab80) as I find it harder to read when actions and assertions are mixed together. This suite actually makes sure the initial value is displayed as expected [here](https://github.com/metabase/metabase/blob/420442d6f5c7e846bb90497d25c284d7563fbec5/frontend/src/metabase/admin/databases/components/DatabaseEditApp/Sidebar/Sidebar.unit.spec.tsx#L177) and [here](https://github.com/metabase/metabase/blob/420442d6f5c7e846bb90497d25c284d7563fbec5/frontend/src/metabase/admin/databases/components/DatabaseEditApp/Sidebar/Sidebar.unit.spec.tsx#L187).

Reply:
That's fair, but do you think you should at least test that after clicking the button the UI changes? Now it only checks that callback is called with the correct arguments.

Reply:
Oh yeah, sorry I missed that, will do!

Reply:
@WiNloSt tried to do that, and now I remember why it wasn't there in the first place. Clicking the toggle calls `updateDatabase` which should change the `database` object. that would lead to toggle `value` change too. Because we're mocking `updateDatabase`, there's no update happening, so we can't really do the assertion

Reply:
Ok that makes sense. Thanks for the explanation!